Do not install when no-install is passed

dwoz opened the pull request:
When trying to stage a release the macos downgrade tests started failing. Looking into why I noticed the package tests were not always doing what we intend. This opened up a bit of a can of worms but I think the PR straightens things out. It fixes the upgrade tests so they are actually performing an upgrade. In addition we're fixing the windows upgrade/downgrade tests.

dwoz committed:
Do not install when no-install is passed

dwoz committed:
Actually run upgrade tests

dwoz committed:
Actually run upgrade

dwoz committed:
Fix bin_file on windows

twangboy commented on tests/pytests/pkg/upgrade/test_salt_upgrade.py:
```diff
@@ -47,6 +45,8 @@ def salt_test_upgrade(
     # Verify previous install version salt-master is setup correctly and works
     bin_file = "salt"
+    if sys.platform == "windows":
+        bin_file = "salt.exe"
```
There is no `salt.exe` on Windows

twangboy replied:
we could probably do `salt-call.exe` instead. I'll try that.

twangboy committed:
Enable WMIC on the runner

dwoz committed:
Add get capabilties

dwoz committed:
Try to get wmic back

dwoz committed:
Windows upgrade test fix-ups

- Work around WMIC disks grain failure by overriding the minion's grain.
- Use static master config directory on windows so it can be re-used by
  the pkg integration tests.

dwoz committed:
Do not clean master key on windows

dwoz committed:
Conf dir is string not tuple

dwoz committed:
Run windows downgrade tests

dwoz committed:
Add comment to help explain what is expected
